tajo-catalog/tajo-catalog-client/src/main/java/org/apache/tajo/catalog/AbstractCatalogClient.java

```diff
@@ -428,13 +429,12 @@ public final PartitionDescProto getPartition(final String databaseName, final St
   }
 
   @Override
-  public final List<PartitionDescProto> getPartitions(final String databaseName, final String tableName) {
+  public final List<PartitionDescProto> getPartitions(final String databaseName, final String tableName) throws
+    UndefinedDatabaseException, UndefinedTableException, UndefinedPartitionMethodException,
```

Comment:
To handle such exceptions, you should put ``throwsIfThisError`` method for the exceptions.